lib/configurators/databases/aws.go

```diff
@@ -0,0 +1,656 @@
+		case types.DatabaseTypeRDS:
+			actions = append(actions, auroraActions...)
+			actions = append(actions, rdsActions...)
+			boundaryActions = append(boundaryActions, boundaryRDSAuroraActions...)
+		case types.DatabaseTypeRedshift:
+			actions = append(actions, redshiftActions...)
+			boundaryActions = append(boundaryActions, boundaryRedshiftActions...)
+		}
+	}
+
+	// If auto-discovery only has "unsupported" databases, return empty
+	// statements.
+	if len(actions) == 0 {
+		return nil, nil
+	}
+
+	switch targetType {
+	case targetTypeUser:
+		actions = append(actions, userBaseActions...)
+	case targetTypeRole:
+		actions = append(actions, roleBaseActions...)
+	}
+
+	policyStatement := &awslib.Statement{Effect: awslib.EffectAllow, Actions: actions, Resources: []string{"*"}}
+	boundaryStatement := &awslib.Statement{Effect: awslib.EffectAllow, Actions: append(actions, boundaryActions...), Resources: []string{"*"}}
+	return policyStatement, boundaryStatement
+}
+
+// buildRedshiftStatements generates policy Statements for Redshift databases.
+func buildRedshiftStatements(fileConfig *config.FileConfig, identity awslib.Identity) ([]*awslib.Statement, error) {
+	accountID := "*"
+	if identity != nil {
+		accountID = identity.GetAccountID()
+	}
+
+	var resources []string
+	for _, database := range fileConfig.Databases.Databases {
+		if strings.Contains(database.URI, types.RedshiftEndpointSuffix) {
+			clusterID, region, err := types.ParseRedshiftEndpoint(database.URI)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			for _, resource := range redshiftResources {
+				resources = append(resources, fmt.Sprintf(
+					"arn:aws:redshift:%s:%s:%s:%s/*",
+					region,
+					accountID,
+					resource,
+					clusterID,
+				))
+			}
+		}
+	}
+
+	if len(resources) == 0 {
+		return []*awslib.Statement{}, nil
+	}
+
+	return []*awslib.Statement{
+		{Effect: awslib.EffectAllow, Actions: []string{"redshift:GetClusterCredentials"}, Resources: resources},
+		{Effect: awslib.EffectAllow, Actions: []string{"redshift:DescribeClusters"}, Resources: []string{"*"}},
+	}, nil
+}
+
+// formatPolicyDocument formats the PolicyDocument in a "friendly" format, which
+// can be presented to end users.
+func formatPolicyDocument(policy *awslib.PolicyDocument) (string, error) {
+	b, err := json.MarshalIndent(policy, "", "    ")
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	return string(b), nil
+}
+
+// encodePolicyDocument encode PolicyDocument into JSON.
+func encodePolicyDocument(policy *awslib.PolicyDocument) (string, error) {
+	b, err := json.Marshal(policy)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	return string(b), nil
+}
+
+// hasTeleportTag check if list of tag contains the Teleport tag.
+func hasTeleportTag(tags []*iam.Tag) bool {
+	for _, tag := range tags {
+		if *tag.Key == policyTeleportTagKey {
+			return true
+		}
+	}
+
+	return false
+}
+
+// wrapAWSError wraps an AWS error accordingly.
+func wrapAWSError(err error) error {
```

Comment:
Is this method needed? Why not just call awslib.ConvertAWSError(e)?

Reply:
It was needed since the `ConvertAWSError` expected an `awserr.RequestFailure`. I've updated the function to receive an `error` and convert internally, so we don't need this function anymore.